Fix smartdown bin script

heathd opened the pull request:
The `bin/smartdown` script is a utility for evaluating smartdown flows from the command line. It's  intended for use by developers when building flows, rather than being used in any production services.

It was untested and as smartdown itself evolved it had broken because there were no tests to reveal that fact.

The actual cause of the breakage was that the `responses` variable was renamed to `accepted_responses` in 1df7819.

This PR fixes the script by changing it to use the correct variable. Also added an integration test.

heathd committed:
Fix smartdown bin script

it had broken because the `responses` variable was renamed to
`accepted_responses` in 1df7819 and there were no tests to reveal this.

Fixed by renaming the variable and adding an integration test.

tadast commented:
LGTM :+1:

jamiecobbett commented on bin/smartdown:
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env ruby
 
+Encoding.default_external = 'UTF-8'
```
This doesn't seem to relate to the commit message - could you split it out and give some context?

jamiecobbett commented on spec/bin/smartdown_spec.rb:
```diff
@@ -0,0 +1,58 @@
+
+  before(:each) do
+    @output = `#{executable_path} "#{input}" #{responses} 2>&1`
+    fail(raw_output) unless $?.success?
```
Can we swap the cryptic `$?` for `$CHILD_STATUS`? You should just need to [`require 'english'`](http://ruby-doc.org/stdlib-2.1.4/libdoc/English/rdoc/English.html) beforehand.

heathd replied:
apparently the file is called `English.rb` not `english.rb`, which matters on case sensitive file systems :-/

heathd committed:
Ensure that tests work in all contexts

in contexts where LANG is not set, the default ruby encoding used for
file I/O would be US-ASCII. This would fail to read the fixture data
used by this test.

Therefore, we explicitly set the LANG environment variable when invoking
the bin/smartdown script.

heathd committed:
Use $CHILD_STATUS instead of $?

it's meaning is more obvious

heathd commented:
@jamiecobbett thanks for reviewing, I didn't know about the more explicit magic variable names available via `require 'english'`.. like :+1: 

updated in line with your comments
